File: 백준/Silver/1929. 소수 구하기/소수 구하기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int M = Integer.parseInt(st.nextToken());
        int N = Integer.parseInt(st.nextToken());

        boolean[] isComposite = new boolean[N + 1]; // false면 소수 후보
        if (N >= 0) isComposite[0] = true;
        if (N >= 1) isComposite[1] = true;

        int limit = (int)Math.sqrt(N);
        for (int p = 2; p <= limit; p++) {
            if (!isComposite[p]) {
                // p*p 미만은 이전에 이미 지워짐
                long start = (long)p * p;
                for (long m = start; m <= N; m += p) {
                    isComposite[(int)m] = true;
                }
            }
        }

        StringBuilder out = new StringBuilder();
        for (int x = Math.max(2, M); x <= N; x++) {
            if (!isComposite[x]) out.append(x).append('\n');
        }
        System.out.print(out.toString());
    }
}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.cc
#include <iostream>
#include <vector>
#include <sstream>
#include <cmath>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int M, N;
    cin >> M >> N;
    vector<bool> isComposite(N+1, false);
    
    if(N >= 0) isComposite[0] = true;
    if(N >= 1) isComposite[1] = true;

    int limit = static_cast<int>(sqrt(N));
    for(int p=2;p<=limit;p++) {
        if(!isComposite[p]) {
            long long start = static_cast<long long>(p*p);
            for(long long m = start;m<=N;m+=p) {
                isComposite[m] = true;
            }
        }
    }

    for(int i=max(2,M);i<=N;i++) {
        if(!isComposite[i]) {
            cout << i << "\n";
        }
    }

    return 0;
}